src/service/models.py

```diff
@@ -0,0 +1,83 @@
+"""
+Pydantic models for the MinIO Manager API.
+"""
+
+from typing import Annotated, Optional, List
+
+from pydantic import BaseModel, Field
+
+
+class ErrorResponse(BaseModel):
+    """Standard error response model."""
+
+    error: Annotated[int | None, Field(description="Error code")] = None
+    error_type: Annotated[str | None, Field(description="Error type")] = None
+    message: Annotated[str | None, Field(description="Error message")] = None
+
+
+class HealthResponse(BaseModel):
+    """Health check response model."""
+
+    status: Annotated[str, Field(description="Health status")]
+
+
+class UserRequest(BaseModel):
+    """Request model for creating a new user."""
+    
+    groups: Annotated[Optional[List[str]], Field(description="List of groups the user belongs to", default=None)]
+
+
+class UserResponse(BaseModel):
+    """Response model for user operations."""
+    
+    username: Annotated[str, Field(description="Username")]
+    access_key: Annotated[str, Field(description="MinIO access key")]
+    secret_key: Annotated[str, Field(description="MinIO secret key")]
+    policy_name: Annotated[str, Field(description="MinIO policy name")]
+    groups: Annotated[List[str], Field(description="List of groups the user belongs to")]
+    home_path: Annotated[str, Field(description="User's home path in MinIO")]
+    status: Annotated[str, Field(description="User status")]
```

Comment:
So we're going to be making a minio user for each kbase user?

Reply:
yea - that's my plan.

Reply:
Hmm, how many users can minio support?

Reply:
What about name collisions, since services and kbase users will be sharing the same namespace?

Reply:
I am thinking we will append an auth group for each minio user, e.g. `kbase_tgu` since we will might support other auth method (BER in the future?). Service accounts such as minio_admin wouldn't have this group appended.

Reply:
Ok, so you'll manually add a namespace and we'll just have to be sure nothing else uses that namespace

:+1: